tests/components/backup/test_manager.py

```diff
@@ -480,6 +481,70 @@ async def test_receive_backup(
     assert remote_agent._backup_data == upload_data.encode(encoding="utf-8")
 
 
+@pytest.mark.usefixtures("mock_backup_generation")
+async def test_receive_backup_busy_manager(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Test receive backup with a busy manager."""
+    assert await async_setup_component(hass, DOMAIN, {})
+    await hass.async_block_till_done()
+    client = await hass_client()
+    ws_client = await hass_ws_client(hass)
+
+    upload_data = "test"
+
+    await ws_client.send_json_auto_id({"type": "backup/subscribe_events"})
+    result = await ws_client.receive_json()
+    assert result["event"] == {"manager_state": "idle"}
+
+    result = await ws_client.receive_json()
+    assert result["success"] is True
+
+    new_backup = NewBackup(backup_job_id="time-123")
+    backup_task: asyncio.Future[WrittenBackup] = asyncio.Future()
+    with patch(
+        "homeassistant.components.backup.manager.CoreBackupReaderWriter.async_create_backup",
+        return_value=(new_backup, backup_task),
+    ) as create_backup:
+        await ws_client.send_json_auto_id(
+            {"type": "backup/generate", "agent_ids": ["backup.local"]}
+        )
+        result = await ws_client.receive_json()
+        assert result["event"] == {
+            "manager_state": "create_backup",
+            "stage": None,
+            "state": "in_progress",
+        }
+        result = await ws_client.receive_json()
+        assert result["success"] is True
+        assert result["result"] == {"backup_job_id": "time-123"}
+
+    assert create_backup.call_count == 1
```

Comment:
This is a lot of code just to make the manager busy IMO and makes the test hard to read.

This is what needs to happen:
```py
    manager = hass.data[DATA_MANAGER]
    manager.last_event = CreateBackupEvent(
        stage=None, state=CreateBackupState.IN_PROGRESS
    )
```

Either simplify like that, or add some test helper to make the manager busy to separate that from what we want to test (how `/api/backup/upload` behaves when the manager is busy)

OK to do that in a follow-up